src/asm/main.cpp

```diff
@@ -376,10 +377,8 @@ int main(int argc, char *argv[])
 		}
 	}
 
-	if (targetFileName == NULL)
-		targetFileName = objectName;
-	else
-		targetFileName[targetFileNameLen - 1] = '\0'; // Overwrite the last space
+	if (!targetFileNames && objectName)
+		targetFileNames = strdup(objectName);
```

Comment:
By using `strdup` instead of assignment, `targetFileNames` is always either allocated or `NULL`, so can be `free`d later; and `objectName` can be `const`.